manifests/functions/create_export.pp

```diff
@@ -38,13 +43,16 @@
       content => $line,
     }
 
+    # Create the directory path only if a File resource isn't
+    #     # defined previously AND the $create_dir boolean is true.
     unless defined(File[$name]) {
-      file { $name:
-        ensure                  => directory,
-        owner                   => $owner,
-        group                   => $group,
-        mode                    => $mode,
-        selinux_ignore_defaults => true,
+      if $create_dir {
+        filepath { $name:
```

Comment:
Why the switch of `file` into `filepath`? That's not relevant when it comes to creating the folder or not and there is another PR for this change.

I would argue that this module almost never should create this folder as that normally is the responsibility of another module. I think it would be even worse to let it create an entire path ...

It would be a breaking change, but I would consider setting $create_dir default to false.

Reply:
Please stay with file for now. Besides this: there are two implementations for creating recursive directories: filepath and dirtrtee. We should have this change in a separate PR so we can check and review which module is the better one to use.

Reply:
And the module it self has this https://github.com/voxpupuli/puppet-nfs/blob/master/manifests/functions/mkdir.pp :)